lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -523,6 +522,28 @@ void ExtractInstancesPass::extractInstances() {
           parent.getArgument(numParentPorts + portIdx));
     }
     assert(inst.use_empty() && "instance ports should have been detached");
+    DenseSet<NonLocalAnchor> instanceNLAs;
+    // Get the NLAs that pass through the InstanceOp `inst`.
+    // This does not returns NLAs that have the `inst` as the leaf.
+    nlaTable.getInstanceNLAs(inst, instanceNLAs);
+    // Map of the NLAs, that are applied to the InstanceOp. That is the NLA
+    // terminates on the InstanceOp.
+    DenseMap<NonLocalAnchor, Annotation> instNonlocalAnnos;
+    // Get the NLAs that are applied on the InstanceOp, since the NLATable does
+    // not return them.
+    AnnotationSet::removeAnnotations(inst, [&](Annotation anno) {
+      auto nlaName = anno.getMember<FlatSymbolRefAttr>("circt.nonlocal");
+      // Ignore any other anno.
+      if (!nlaName)
+        return false;
+      // Ignore the breadcrumb annos, since NLATable already tracks them.
+      if (!anno.isClass("circt.nonlocal"))
```

Comment:
Would it make sense to ensure that we never create breadcrumb annos instead of actively removing them here? I suspect that there will be other passes looking at NLAs on instances, and we probably don't want to have each get rid of any stray breadcrumbs.